var/spack/repos/builtin/packages/cbtf-argonavis/package.py

```diff
@@ -67,6 +67,62 @@ class CbtfArgonavis(Package):
 
     parallel = False
 
+    build_directory = 'build_cbtf_argonavis'
+
+    # We have converted from Package type to CMakePackage type for all the Krell projects.
+    # Comments from Pull Request (#4765):
+    # CMakePackage is completely different from the old Package class. Previously, you had 
+    # a single install() phase to override. Now you have 3 phases: cmake(), build(), and install(). 
+    # By default, cmake() runs cmake ... with some common arguments, which you can add to by 
+    # overriding cmake_args(). build() runs make, and install() runs make install. 
+    # So you need to add the appropriate flags to cmake_args() and remove the calls to make. 
+    # See any other CMakePackage for examples.
+    # CMakePackage is documented: 
+    # http://spack.readthedocs.io/en/latest/spack.build_systems.html?highlight= \
+    # CMakePackage#module-spack.build_systems.cmake
+
+    def build_type(self):
+        if '+debug' in self.spec:
+            return 'Debug'
+        else:
+            return 'Release'
+
+    def cmake_args(self):
+        spec = self.spec
+
+        # Look for package installation information in the cbtf and cbtf-krell
+        # prefixes
+        cmake_prefix_path = join_path(
+            spec['cbtf'].prefix) + ':' + join_path(spec['cbtf-krell'].prefix)
+
+        cmake_args = []
+        cmake_args.extend(
+            ['-DCMAKE_INSTALL_PREFIX=%s' % prefix,
+             '-DCMAKE_PREFIX_PATH=%s' % cmake_prefix_path,
+             '-DCUDA_DIR=%s' % spec['cuda'].prefix,
+             '-DCUDA_INSTALL_PATH=%s' % spec['cuda'].prefix,
+             '-DCUDA_TOOLKIT_ROOT_DIR=%s' % spec['cuda'].prefix,
+             '-DCUPTI_DIR=%s' % join_path(
+                 spec['cuda'].prefix + '/extras/CUPTI'),
+             '-DCUPTI_ROOT=%s' % join_path(
+                 spec['cuda'].prefix + '/extras/CUPTI'),
+             '-DPAPI_ROOT=%s' % spec['papi'].prefix,
+             '-DCBTF_DIR=%s' % spec['cbtf'].prefix,
+             '-DCBTF_KRELL_DIR=%s' % spec['cbtf-krell'].prefix,
+             '-DBOOST_ROOT=%s' % spec['boost'].prefix,
+             '-DBoost_DIR=%s' % spec['boost'].prefix,
+             '-DBOOST_LIBRARYDIR=%s' % spec['boost'].prefix.lib,
+             '-DMRNET_DIR=%s' % spec['mrnet'].prefix,
+             '-DBoost_NO_SYSTEM_PATHS=ON'])
+
+        # Add in the standard cmake arguments
+        cmake_args.extend(std_cmake_args)
+
+        # Adjust the standard cmake arguments to what we want the build
+        # type, etc to be
+        self.adjustBuildTypeParams_cmakeOptions(spec, cmake_args)
```

Comment:
You don't need to adjust the build type anymore. It is already covered by the `build_type` function.